packages/malloy-db-bigquery/src/bigquery_connection.ts

```diff
@@ -210,15 +216,42 @@ export class BigQueryConnection extends Connection {
         ? jobResult[2].totalRows
         : "0");
 
+      if (jobResult[2]?.schema === undefined) {
+        throw new Error("Schema not present");
+      }
+
       // TODO even though we have 10 minute timeout limit, we still should confirm that resulting metadata has "jobComplete: true"
-      result = { rows: jobResult[0], totalRows };
-      this.resultCache.set(hash, result);
-      return result;
+      const data = { rows: jobResult[0], totalRows };
+      const schema = jobResult[2]?.schema;
+
+      this.resultCache.set(hash, { data, schema });
+      return { data, schema };
     } catch (e) {
       throw maybeRewriteError(e);
     }
   }
 
+  public async runSQL(
+    sqlCommand: string,
+    options: Partial<BigQueryQueryOptions> = {},
+    rowIndex = 0
+  ): Promise<MalloyQueryData> {
+    const { data } = await this._runSQL(sqlCommand, options, rowIndex);
+    return data;
+  }
+
+  public async runSQLBlockAndFetchResultSchema(
```

Comment:
@bporterfield I've called it `runSQLBlockAndFetchResultSchema` for now... Because the most useful method is to run specifically a SQL block, and not just any old SQL. Not too sure if this is the right API though.

Reply:
seems good enough 👍